engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -101,13 +104,13 @@ public T getValue() {
     }
 
     public boolean setValue(T newValue) {
-        if (!validator.validate(newValue))
+        if (!validator.validate(newValue)) {
+            LOGGER.warn("The passed value {} is invalid.", newValue);
```

Comment:
That's why I was asking. Do you want the `validate` method to log a warning itself or do you want another method that encapsulates `validate` and logs a warning if `validate` returns false? Or maybe we can pass in a flag to `validate` that states whether it should log a warning or not. Personally I wouldn't log a warning by _default_ since you wouldn't expect a validating method to log a warning if the value is invalid since its job is just telling you whether the value is valid or not.